module/mempool/chunk_requests.go

```diff
@@ -1,17 +1,67 @@
 package mempool
 
 import (
+	"time"
+
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/model/verification"
 )
 
+// ChunkRequestHistoryUpdaterFunc is a function type that used by ChunkRequests mempool to perform atomic and isolated updates on the
```

Comment:
@jordanschalm thanks for your comment, the logic for this injection was to make the maintenance and updating of `ChunkRequest` data object separate from each other. In this way, `ChunkRequests` mempool is not integrated with the updating logic of requests. Rather the logic is _injected_ and the mempool is only responsible for _executing_ the injected logic. In this way, the mempool is only responsible for maintaining the requests on storage and executing some injectable logic. But the mempool itself is not responsible for _maintaining the logic_ itself. It makes it easier to plug-and-play new `updaters` due to reasons like benchmarking, firefighting, etc. I agree that we are hoping to eventually use the `ExponentialUpdater`, but we are not yet certain that this is the performant one. There are still some ambiguities around it that make clear once we roll into production, e.g., we may aggressively hit the cutoff due to slow response of the execution nodes and may want to try another updater behavior, for example, geometric one. Then instead of refactoring an integrated logic, we just need to implement the new one succinctly into a new updater function and inject that one.   

Please let me know your thoughts on this.